bert-burn/src/model.rs

```diff
@@ -47,12 +50,41 @@ pub struct BertModelConfig {
 pub struct BertModel<B: Backend> {
     pub embeddings: BertEmbeddings<B>,
     pub encoder: TransformerEncoder<B>,
+    pub pooler: Option<Pooler<B>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct BertModelOutput<B: Backend> {
+    pub hidden_states: Tensor<B, 3>,
+    pub pooled_output: Option<Tensor<B, 3>>,
 }
 
 impl BertModelConfig {
     /// Initializes a Bert model with default weights
-    pub fn init<B: Backend>(&self, device: &B::Device) -> BertModel<B> {
-        let embeddings = BertEmbeddingsConfig {
+    pub fn init<B: Backend>(&self, device: &B::Device, with_pooling_layer: bool) -> BertModel<B> {
```

Comment:
I think I would put the `pooling_layer` in the config.

Reply:
:+1: I went back and forth on it, and I ended up going in this direction because I reasoned that this was a Burn-specific modifier rather than an actual model config parameter. I can see how it's more obvious in the config, though, so that distinction doesn't really matter. :smile: I'll make the change, and stage the PR against the tracel-ai repo for final review. Thanks for taking a look!